samples/spring-security-xsuaa-usage/src/main/resources/_vcap-services.json

```diff
@@ -0,0 +1,57 @@
+{
+    "xsuaa": [
```

Comment:
Actually, the API of the PR is not limited to a VCAP_SERVICES environment.

The API exposes a bean of type `XsuaaServiceBindings`, `XsuaaBindingInfo` and `Credentials` - which are interfaces simply providing getters to binding information.  

The *implementation* of the API is bound to VCAP_SERVICES right now, since that is what applications building on SAP CP CF need right now. 
In case you wanted to get the information exposed by the interfaces mentioned above in a K8s environment, you could always provide a K8s-based implementation and expose the `XsuaaServiceBindings` bean with an '@Conditional' annotation that checks in which environment the application is currently running.  
An application would not have to be changed in such a case.

The downside with the current *.properties-based implementation is that it has no support for multiple XSUAA instances. And it uses a custom property path that is not "codified" but can only be documented. I exposed a bean on purpose, since it provides an API to an application - one that is similar in structure to what an application sees in `VCAP_SERVICES` but is not limited to Cloud Foundry.